python/lsst/pipe/tasks/makeDirectWarp.py

```diff
@@ -0,0 +1,799 @@
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)  # Harmless and a no-op for now.
+        if not config:
+            return
+
+        if config.bgSubtracted:
+            del self.background_list
+        if not config.doApplySkyCorr:
+            del self.skyCorr_list
+
+        # Dynamically set output connections for noise images, depending on the
+        # number of noise realization specified in the config.
+        for n in range(config.numberOfNoiseRealizations):
+            noise_warp = Output(
+                doc="Output direct warped noise exposure",
+                name=f"{config.connections.coaddName}Coadd_directWarp_noise{n}",
+                # Store it as a MaskedImage to preserve the variance plane.
+                storageClass="MaskedImageF",
+                dimensions=("tract", "patch", "skymap", "instrument", "visit"),
+            )
+            setattr(self, f"direct_noise_warp{n}", noise_warp)
+
+
+class MakeDirectWarpConfig(
+    PipelineTaskConfig,
+    pipelineConnections=MakeDirectWarpConnections,
+):
+    """Configuration for the MakeDirectWarpTask.
+
+    The config fields are as similar as possible to the corresponding fields in
+    MakeWarpConfig.
+
+    Notes
+    -----
+    The config fields are in camelCase to match the fields in the earlier
+    version of the makeWarp task as closely as possible.
+    """
+
+    MAX_NUMBER_OF_NOISE_REALIZATIONS = 3
+    """
+    numberOfNoiseRealizations is defined as a RangeField to prevent from
+    making multiple output connections and blowing up the memory usage by
+    accident. An upper bound of 3 is based on the best guess of the maximum
+    number of noise realizations that will be used for metadetection.
+    """
+
+    numberOfNoiseRealizations = RangeField[int](
+        doc="Number of noise realizations to simulate and persist.",
+        default=2,
+        min=0,
+        max=MAX_NUMBER_OF_NOISE_REALIZATIONS,
+        inclusiveMax=True,
+    )
+    seedOffset = Field[int](
+        doc="Offset to the seed used for the noise realization.",
+        default=0,
+    )
+    useMedianVariance = Field[bool](
+        doc="Use the median of variance plane in the input calexp to generate "
+        "noise realizations? If False, per-pixel variance will be used.",
+        default=True,
+    )
+    bgSubtracted = Field[bool](
+        doc="Work with a background subtracted calexp?",
+        default=True,
+    )
+    doApplySkyCorr = Field[bool](
+        doc="Apply sky correction?",
+        default=False,
+    )
+    useVisitSummaryPsf = Field[bool](
+        doc="If True, use the PSF model and aperture corrections from the 'visitSummary' connection. "
+        "If False, use the PSF model and aperture corrections from the 'calexp' connection.",
+        default=True,
+    )
+    badMaskPlanes = ListField[str](
+        doc="Mask planes that count toward the masked fraction and "
```

Comment:
You're right. In `makeWarp`, this was used to determine which pixels to mark as bad ones and not copy over to the final warp. But given that we do an interpolation, it can be hardcoded to be `NO_DATA` in the code directly.